mantidproject/mantid
# Muon multiple fitting gui

This is the first phase of an upgrade to the multiple fitting GUI in muon analysis to make it more user friendly. 

If a single fit (one period and one group/pair) is done then the labels are wrong and already has an issue #19544 

**To test:**


There are multiple tests. I recommend using HIFI data (never use the current run) as it has both multiple periods and groups. 

0. The input parameters change after a simultaneous fit if a different data set is selected for display (assuming it is not global). 

1. Changing the Groups/Pairs to fit option updates the GUI correctly. So the relevant data should be listed in the Display parameters for drop down menu. A fit should use the relevant data (can check by doing a tiled plot of the output workspace). 

2. Changing the periods to fit option updates the GUI correctly. The response should be similar to the above. 

3. If custom is selected from the drop down menus (in 1 and 2) then the relevant pop up is produced and data is passed to the GUI correctly. If you press the Reselect Groups/Pairs or Periods buttons (will only be enabled if the relevant drop down menu is set to custom) the pop up menu should reappear. 

4. If the data has multiple periods, the Combine Periods button will produce a pop up that allows users to create combinations of periods. This should be added to the period drop down and pop ups. Check that the data is passed to the GUI correctly. 

5. Changing the start and end alters the fit. 

6. User can switch between multi and single fit options and the fits work (numbers don't matter).

7. Single fit produces the same answers as before. If you change the workspace in data analysis there is a bug #19571 

Fixes #19149  and #19544 



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
